frontend/views/containers/chatroom/MessageActions.vue

```diff
@@ -130,6 +133,27 @@ import { MenuParent, MenuTrigger, MenuContent, MenuItem } from '@components/menu
 import Tooltip from '@components/Tooltip.vue'
 import { MESSAGE_TYPES, MESSAGE_VARIANTS } from '@model/contracts/shared/constants.js'
 
+const getAvailableMoreOptions = function () {
+  const moreOptions = [{
+    name: 'Copy message text',
+    conditionToShow: this.isText
+  }, {
+    name: 'Copy message link',
+    conditionToShow: true
+  }, {
+    name: 'Pin to channel',
+    conditionToShow: !this.isAlreadyPinned && this.isPinnable
+  }, {
+    name: 'Unpin from channel',
+    conditionToShow: this.isAlreadyPinned
+  }, {
+    name: 'Delete message',
+    conditionToShow: this.isDeletable
+  }]
+
+  return moreOptions.filter(option => option.conditionToShow)
+}
```

Comment:
This is a cool idea 👍 

One thing though: remember all user-facing strings should be wrapped in either `L` function or `i18n`. Here the `L` function would be needed.

Reply:
@Silver-IT Agree this is a tidy solution. Since all `this.something` here look like the component props, I think this list can be stored as a computed prop instead, so that `conditionToShow` here are evaluated one time and the results are cached.

Reply:
The function `getAvailableMoreOptions` was not used in HTML, so I didn't wrapped name with `L`. 
But I will make computed value `moreOptions` (instead of `getAvailableMoreOptions`) and will wrap `name` with `L` and use it in HTML and `moreOptionsTriggered` function too.
That will fix the DRY too.